community/lucene-index/src/test/java/org/neo4j/kernel/impl/api/OperationsFacadeSchemaIndexIteratorTest.java

```diff
@@ -1,766 +0,0 @@
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.kernel.impl.api;
-
-import org.junit.Assume;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.RuleChain;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
-import org.neo4j.collection.primitive.PrimitiveLongCollections;
-import org.neo4j.collection.primitive.PrimitiveLongIterator;
-import org.neo4j.graphdb.Label;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Transaction;
-import org.neo4j.internal.kernel.api.IndexQuery;
-import org.neo4j.kernel.api.ReadOperations;
-import org.neo4j.kernel.api.Statement;
-import org.neo4j.kernel.api.exceptions.index.IndexNotApplicableKernelException;
-import org.neo4j.kernel.api.exceptions.index.IndexNotFoundKernelException;
-import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptor;
-import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptorFactory;
-import org.neo4j.test.rule.DatabaseRule;
-import org.neo4j.test.rule.EmbeddedDatabaseRule;
-import org.neo4j.test.rule.RandomRule;
-
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
-import static org.junit.Assert.assertThat;
-
-/**
- * Does test having multiple iterators open on the same index
- * <ul>
- * <li>Exhaust variations:</li>
- * <ul>
- * <li>Exhaust iterators one by one</li>
- * <li>Nesting</li>
- * <li>Interleaved</li>
- * </ul>
- * <li>Happy case for schema index iterators on static db for:</li>
- * <ul>
- * <li>Single property number index</li>
- * <li>Single property string index</li>
- * <li>Composite property number index</li>
- * <li>Composite property string index</li>
- * </ul>
- * <li>For index queries:</li>
- * <ul>
- * <li>{@link IndexQuery#exists(int)}</li>
- * <li>{@link IndexQuery#exact(int, Object)}</li>
- * <li>{@link IndexQuery#range(int, Number, boolean, Number, boolean)}</li>
- * <li>{@link IndexQuery#range(int, String, boolean, String, boolean)}</li>
- * </ul>
- * </ul>
- * Does NOT test
- * <ul>
- * <li>Single property unique number index</li>
- * <li>Single property unique string index</li>
- * <li>Composite property mixed index</li>
- * <li>{@link IndexQuery#stringPrefix(int, String)}</li>
- * <li>{@link IndexQuery#stringSuffix(int, String)}</li>
- * <li>{@link IndexQuery#stringContains(int, String)}</li>
- * <li>Composite property node key index (due to it being enterprise feature)</li>
- * <li>Label index iterators</li>
- * <li>Concurrency</li>
- * <li>Locking</li>
- * <li>Cluster</li>
- * <li>Index creation</li>
- * </ul>
- * Code navigation:
- */
-@RunWith( Parameterized.class )
-public class OperationsFacadeSchemaIndexIteratorTest
```

Comment:
This seems like a useful test, but that could potentially move to the `IndexProvider` level.

Reply:
migrated